src/components/cylc/graph/Graph.vue

```diff
@@ -0,0 +1,631 @@
+            }
+            content.innerHTML += `<hr class="hr-graph"><div class="header-graph"><strong>jobs</div></strong><br>${jobBlock}`
+            return content
+          },
+          theme: 'light-border',
+          trigger: 'manual',
+          arrow: true,
+          placement: 'left',
+          hideOnClick: 'true',
+          delay: [0, 2000],
+          animation: 'fade',
+          multiple: false,
+          sticky: true,
+          flip: true,
+          flipOnUpdate: true,
+          duration: [250, 275],
+          allowHTML: true,
+          interactive: true,
+          touch: true
+        })
+        this.tippy.show()
+      })
+
+      instance.on('tap', 'edge', (event) => {
+        const edge = event.target
+        edge.addClass('selected')
+        this.tippy = new Tippy(edge.popperRef(), {
+          content: () => {
+            const content = document.createElement('div')
+            content.classList.add('edge-tooltip')
+            content.innerHTML = `
+              <div class="header-graph">
+                <strong>edge</div></strong>
+                <br/>
+                <strong>owner: <span class="details-graph">${edge.data('owner')}</span></strong>
+                <br/>
+                <strong>name: <span class="details-graph">${edge.data('name')}</span></strong>
+                <br/>
+                <strong>source: <span class="details-graph">${edge.data('source')}</span></strong>
+                <br/>
+                <strong>target: <span class="details-graph">${edge.data('target')}</span></strong>
+                <br/>
+                <strong>label: <span class="details-graph">${edge.data('label')}</span></strong>
+              </div>`
+            return content
+          },
+          allowHTML: true,
+          animation: 'fade',
+          arrow: true,
+          delay: [0, 2000],
+          duration: [250, 275],
+          flip: true,
+          flipOnUpdate: true,
+          hideOnClick: 'false',
+          interactive: true,
+          multiple: false,
+          placement: 'left',
+          sticky: true,
+          theme: 'light-border',
+          trigger: 'manual'
+        })
+        this.tippy.show()
+      })
+
+      instance.on('click', (_) => {
+        this.cytoscapeInstance.elements().removeClass('semitransp')
+        this.cytoscapeInstance.elements().removeClass('highlight')
+        this.cytoscapeInstance.elements().removeClass('selected')
+        if (this.tippy !== null) {
+          this.tippy.hide()
+        }
+      })
+    },
+
+    /**
+     * Sets up the event listeners for the instance.
+     * NB: remember to remove the event listeners to avoid memory leaks.
+     */
+    setupEventListeners () {
+      document.addEventListener('keydown', this.keydownEventHandler)
+    },
+
+    keydownEventHandler (event) {
+      if ((event.metaKey && event.key === 'Z') || (event.ctrlKey && event.key === 'Z')) {
```

Comment:
(`which` is deprecated - https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which, `key` is a possible replacement... TL;DR: `which` behaviour varies per browser)

Reply:
Vue has some support for modifier keys too https://vuejs.org/v2/guide/events.html#System-Modifier-Keys